### PR TITLE
Added localization for iOS plist files

### PR DESCRIPTION
**Pull request by enahum**
#### Summary
Permission dialogs on iOS use native mechanism to apply localization to usage description strings in them. This PR adds ability to populate native iOS translation resource files with translated values according to [Apple guide](https://developer.apple.com/library/archive/documentation/General/Reference/InfoPlistKeyReference/Articles/AboutInformationPropertyListFiles.html). This is also applicable for other localizable strings in Info.plist file.

- Updated the lane 'ios replace_assets' in Fastlane file with script, which finds and copies the iOS-specific localization strings into iOS project and replaces the `{applicationName}` with actual value.
- Added new strings to en.json file, based on the strings from `ios/Mattermost/Info.plist`
- iOS project file updated to include translation resource files (for all currently supported languages in Mattermost project), so they will be included to the app package.

#### Ticket
https://mattermost.atlassian.net/browse/MM-52217

#### Checklist
- [ ] Added or updated unit tests (required for all new features)
- [ ] Has UI changes
- [x] Includes text changes and localization file updates
- [ ] Have tested against the 5 core themes to ensure consistency between them.

#### Device Information
This PR was tested on: iPhone simulator, iOS 16.2

#### Release Note
```release-note
* Added localization support for iOS usage description strings on permission dialogs.
```

Opening this PR in favor of #7301

### CODE REVIEW

File: app/constants/permissions.ts [hunk 1]
```diff
@@ -1,5 +1,57 @@
...
+        id: t('mobile.ios.plist.NSLocationAlwaysUsageDescription'),
+        defaultMessage: 'Enabling access to your location data means we can add location metadata to pictures and videos you share in {applicationName}.',
+    },
+    NSLocationWhenInUseUsageDescription: {
+        id: t('mobile.ios.plist.NSLocationWhenInUseUsageDescription'),
+        defaultMessage: 'Enabling access to your location data means we can add location metadata to pictures and videos you share in {applicationName}.',
+    },
+    NSMicrophoneUsageDescription: {
+        id: t('mobile.ios.plist.NSMicrophoneUsageDescription'),
+        defaultMessage: 'Enabling access to your device microphones means you can capture call or video to share in {applicationName}.',
```
Comment (cpoile) [inline app/constants/permissions.ts]:
```suggestion
        defaultMessage: 'Enabling access to your device\'s microphones means you can capture audio for calls or videos to share in {applicationName}.',
```

File: app/constants/permissions.ts [hunk 2]
```diff
@@ -1,5 +1,57 @@
...
+        id: t('mobile.ios.plist.NSAppleMusicUsageDescription'),
+        defaultMessage: 'Enabling access to your media library means you can attach files from your media library to your messages in {applicationName}.',
+    },
+    NSBluetoothAlwaysUsageDescription: {
+        id: t('mobile.ios.plist.NSBluetoothAlwaysUsageDescription'),
+        defaultMessage: 'Enabling access to Bluetooth means we can synchronize content across your devices and clients.',
+    },
+    NSBluetoothPeripheralUsageDescription: {
+        id: t('mobile.ios.plist.NSBluetoothPeripheralUsageDescription'),
+        defaultMessage: 'Enabling access to Bluetooth means we can synchronize content across your devices and clients.',
```
Comment (cpoile) [inline app/constants/permissions.ts:19]:
```suggestion
        defaultMessage: 'Enabling access to Bluetooth means we can connect to audio peripherals for calls, and synchronize content across your devices and clients.',
```

File: assets/base/i18n/en.json [hunk 3]
```diff
@@ -539,6 +538,17 @@
   "mobile.integration_selector.loading_options": "Loading options...",
   "mobile.ios.photos_permission_denied_description": "Upload photos and videos to your server or save them to your device. Open Settings to grant {applicationName} Read and Write access to your photo and video library.",
   "mobile.ios.photos_permission_denied_title": "{applicationName} would like to access your photos",
+  "mobile.ios.plist.NSAppleMusicUsageDescription": "Enabling access to your media library means you can attach files from your media library to your messages in {applicationName}.",
+  "mobile.ios.plist.NSBluetoothAlwaysUsageDescription": "Enabling access to Bluetooth means we can synchronize content across your devices and clients.",
+  "mobile.ios.plist.NSBluetoothPeripheralUsageDescription": "Enabling access to Bluetooth means we can synchronize content across your devices and clients.",
```
Comment (cpoile) [inline assets/base/i18n/en.json:543]:
```suggestion
  "mobile.ios.plist.NSBluetoothPeripheralUsageDescription": "Enabling access to Bluetooth means we can connect to audio peripherals for calls, and synchronize content across your devices and clients.",
```

File: app/constants/permissions.ts (hunk 1 above)
Comment (cwarnermm) [inline app/constants/permissions.ts:39]:
```suggestion
        defaultMessage: 'Enabling access to your device microphones means you can capture call or video audio to share in {applicationName}.',
```

File: assets/base/i18n/en.json [hunk 4]
```diff
@@ -539,6 +538,17 @@
...
+  "mobile.ios.plist.NSCameraUsageDescription": "Enabling access to your device cameras means you can take photos or videos and upload them to {applicationName}.",
+  "mobile.ios.plist.NSFaceIDUsageDescription": "Enabling access to your Face ID means we can restrict unauthorized users from accessing {applicationName} on your device.",
+  "mobile.ios.plist.NSLocationAlwaysUsageDescription": "Enabling access to your location data means we can add location metadata to pictures and videos you share in {applicationName}.",
+  "mobile.ios.plist.NSLocationWhenInUseUsageDescription": "Enabling access to your location data means we can add location metadata to pictures and videos you share in {applicationName}.",
+  "mobile.ios.plist.NSMicrophoneUsageDescription": "Enabling access to your device microphones means you can capture call or video to share in {applicationName}.",
```
Comment (cpoile) [inline assets/base/i18n/en.json]:
```suggestion
  "mobile.ios.plist.NSMicrophoneUsageDescription": "Enabling access to your device\'s microphones means you can capture audio for calls or videos to share in {applicationName}",
```

Comment (cwarnermm) [inline assets/base/i18n/en.json:548]:
```suggestion
  "mobile.ios.plist.NSMicrophoneUsageDescription": "Enabling access to your device microphones means you can capture call or video audio to share in {applicationName}.",
```

File: ios/Mattermost/Info.plist [hunk 5]
```diff
@@ -58,31 +58,27 @@
...
-	<string>Send your current location to $(PRODUCT_NAME)</string>
+	<string>Enabling access to your location data means we can add location metadata to pictures and videos you share in $(PRODUCT_NAME).</string>
 	<key>NSLocationWhenInUseUsageDescription</key>
-	<string>Send your current location to $(PRODUCT_NAME)</string>
+	<string>Enabling access to your location data means we can add location metadata to pictures and videos you share in $(PRODUCT_NAME).</string>
 	<key>NSMicrophoneUsageDescription</key>
-	<string>Capture audio when making a call or recording a video to share within $(PRODUCT_NAME)</string>
-	<key>NSMotionUsageDescription</key>
-	<string>Share your route in your $(PRODUCT_NAME) workspace</string>
+	<string>Enabling access to your device microphones means you can capture call or video to share in $(PRODUCT_NAME).</string>
```
Comment (cwarnermm) [inline ios/Mattermost/Info.plist:75]:
```suggestion
	<string>Enabling access to your device microphones means you can capture call or video audio to share in $(PRODUCT_NAME).</string>
```

**Review by cwarnermm — APPROVED** [review]

File: ios/Mattermost/Info.plist (hunk 5 above)
Comment (cpoile) [inline ios/Mattermost/Info.plist]:
```suggestion
	<string>Enabling access to your device's microphones means you can capture audio for calls or videos to share in $(PRODUCT_NAME)</string>
```

File: ios/Mattermost/Info.plist [hunk 6]
```diff
@@ -58,31 +58,27 @@
...
-	<string>Let $(PRODUCT_NAME) access your Media files</string>
+	<string>Enabling access to your media library means you can attach files from your media library to your messages in $(PRODUCT_NAME).</string>
 	<key>NSBluetoothAlwaysUsageDescription</key>
-	<string>Share post data across devices with $(PRODUCT_NAME)</string>
+	<string>Enabling access to Bluetooth means we can synchronize content across your devices and clients.</string>
 	<key>NSBluetoothPeripheralUsageDescription</key>
-	<string>Share post data across devices with $(PRODUCT_NAME)</string>
-	<key>NSCalendarsUsageDescription</key>
-	<string>Share contacts within $(PRODUCT_NAME)</string>
+	<string>Enabling access to Bluetooth means we can synchronize content across your devices and clients.</string>
```
Comment (cpoile) [inline ios/Mattermost/Info.plist:65]:
```suggestion
	<string>Enabling access to Bluetooth means we can connect to audio peripherals for calls, and synchronize content across your devices and clients.</string>
```

**Review by cpoile — APPROVED** [review]
Just minor suggestions for changing the microphone and bluetooth strings that Calls uses -- use it if you'd like. (I think I got them all, but you may need to regenerate.)

**Comment by enahum** [comment]
@cwarnermm which suggestion should we use the one you added or @cpoile 's ?

**Comment by cwarnermm** [comment]
Let's go with @cpoile's updates. Thanks, @cpoile!

**Comment by enahum** [comment]
@cwarnermm ok I'll apply the changes you missed for the mic

**Pull request merged**